packages/events-api/src/adapter.ts

```diff
@@ -0,0 +1,156 @@
+import EventEmitter from 'events'; // tslint:disable-line
+import http, { IncomingMessage, ServerResponse } from 'http';
+import debugFactory from 'debug'; // tslint:disable-line
+import { createHTTPHandler } from './http-handler';
+
+const debug = debugFactory('@slack/events-api:adapter');
+
+/**
+ * An adapter for Slack's Events API.
+ */
+export class SlackEventAdapter extends EventEmitter {
+  /**
+   * The token used to authenticate signed requests from Slack's Events API.
+   */
+  public readonly signingSecret: string;
+
+  /**
+   * Whether to include the API event bodies in adapter event consumers.
+   */
+  public includeBody: boolean;
+
+  /**
+   * Whether to include request headers in adapter event consumers.
+   */
+  public includeHeaders: boolean;
+
+  /**
+   * When `true`, prevents the adapter from responding by itself and leaves that up to consumers.
+   */
+  public waitForResponse: boolean;
+
+  /**
+   * The HTTP server this adapter might be running, created in {@link start}.
+   */
+  private server?: http.Server;
+
+  /**
+   * @param signingSecret - The token used to authenticate signed requests from Slack's Events API.
+   * @param opts.includeBody - TODO:
+   * @param opts.includeHeaders - TODO:
+   * @param opts.waitForResponse - TODO:
+   */
+  constructor(
+    signingSecret: string | String,
```

Comment:
The current JS version accepted strings *and* `String`s. `new String('x')` would make it through the current "type check" and be used because [`lodash.isstring`](https://github.com/lodash/lodash/blob/master/isString.js#L20) would check if the argument's type was a string or (essentially) if it was an instance of `String`. This typing reflects that.

Reply:
I see your point. My personal opinion is the TypeScript interfaces to public API should be written to deliver the most sensible experience to another TypeScript author (and that will 99% of the time also give the sensible experience to a JavaScript author).

The actual implementation might allow for different types, but that always has been and will be the reality: values that behave enough like the type expected will pass our runtime checks. Therefore I think we should declare the type as `string` (as this communicates the intention) and still validate through `lodash.isstring` (as this enforces the exact same constraints as before, which holds the API from breaking).